### ISSUE TITLE
add support for max_retries_down of openstack_lb_monitor_v2

### ISSUE THREAD

**Issue by f4rx** [terraform-providers/terraform-provider-openstack#805]
Octavia load-balancer API has the option   max-retries-down
```bash
openstack loadbalancer healthmonitor  create --help
...
--max-retries-down <max_retries_down>
                        Set the number of allowed check failures before
                        changing the operating status of the member to ERROR.
```

It would be great if this option is supported in lb_monitor_v2

https://www.terraform.io/docs/providers/openstack/r/lb_monitor_v2.html

**Comment by kayrus** [comment]
Current terraform openstack provider lbaas implementation is based on lbaas v2 extension, but the requested feature is supported in octavia only. @jtopjian if i'm not mistaken, you had an idea on how to introduce the octavia interface. Could you please link related github issues?

**Comment by ozerovandrei** [comment]
@kayrus we have support for the Octavia API and there is also a separate boolean flag to turn this on: [use_octavia](https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/provider.go#L193).

**Comment by kayrus** [comment]
@ozerovandrei unfortunately it is only for the `gophercloud.ServiceClient` struct only, see https://github.com/terraform-providers/terraform-provider-openstack/blob/77673807914733f1ebe536aafb7521c8189f2555/openstack/lb_v2_shared.go#L30

There should be a logic to use neutron extension lib vs octavia lib, not only the client struct.

**Comment by jtopjian** [comment]
It should be possible to add a new schema item called `max_retries_down` and only pass the value in the request body if `config.UseOctavia` is set.

Right now `config.UseOctavia` is only used to create a service client, but there shouldn't be anything preventing it from being used as a conditional in other areas. The only downside is that the load balancer resources will begin to exhibit different behavior depending on if it's set. However, that's probably still less effort than creating a whole new suite of dedicated octavia-based resources. At least for now. 🙂

**Comment by ozerovandrei** [comment]
Implemented in #945 and will be available in the next release.
